nixos/modules/services/matrix/matrix-sliding-sync.nix

```diff
@@ -0,0 +1,63 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.matrix-synapse.sliding-sync;
+in
+{
+  options.services.matrix-synapse.sliding-sync = {
+    enable = lib.mkEnableOption (lib.mdDoc "Sliding Sync Proxy");
+
+    package = lib.mkPackageOption pkgs "matrix-sliding-sync" { };
+
+    listenAddress = lib.mkOption {
+      type = lib.types.str;
+      default = "127.0.0.1";
+      description = "The address sliding sync proxy should listen.";
+    };
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 8449;
+      description = lib.mdDoc ''
+        The port under which the sliding sync proxy should listen.
+      '';
+    };
+
+    environmentFile = lib.mkOption {
```

Comment:
I would prefer a dedicated secret file specifically for `SYNCV3_SECRET` instead.
Or even dropping this altogether in favor of creating that secret dynamically on first start.

I might do a follow-up PR for that some time in the future

Reply:
hmm, i would prefer to at least have a built-in way to specify a secret file; for migrating etc. I prefer to always have all secrets deployed, so there is no state (except database) on the systems which is needed for running